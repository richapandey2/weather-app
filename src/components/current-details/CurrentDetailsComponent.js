import React from "react";
import { Box, makeStyles } from "@material-ui/core";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { convertFahrenheittoCelcius } from "../../utils/formatter";
import { CELCIUS_ICON, FAHRENHEIT_ICON } from "../../constants/Constants";
import "../../Styles.css";

function CurrentDetailsComponent(props) {
  const useStyle = makeStyles(() => ({
    box: {
      display: "flex",
      justifyContent: "center",
      padding: "30px",
    },
  }));
  const classes = useStyle();
  const { overAllDetails, isCelcius } = props;
  return (
    <>
      {overAllDetails && (
        <Box className={classes.box}>
          <div className="main-div">
            <div className="primary-text">
              {overAllDetails.city},{overAllDetails.country}
            </div>
            <div className="temp-text">
              {isCelcius
                ? convertFahrenheittoCelcius(overAllDetails.temp) + CELCIUS_ICON
                : overAllDetails.temp + FAHRENHEIT_ICON}
            </div>
            <div className="secondary-text">{overAllDetails.date}</div>
          </div>
        </Box>
      )}
    </>
  );
}

CurrentDetailsComponent.propTypes = {
  overAllDetails: PropTypes.shape({
    date: PropTypes.string,
    temp: PropTypes.number,
    weather: PropTypes.string,
    icon: PropTypes.string,
    country: PropTypes.string,
    city: PropTypes.string,
  }),
  isCelcius: PropTypes.bool,
};

const mapStateToProps = (state) => {
  return {
    overAllDetails: state.weatherReducer.data.overAllDetails,
  };
};

export default connect(mapStateToProps, null)(CurrentDetailsComponent);
