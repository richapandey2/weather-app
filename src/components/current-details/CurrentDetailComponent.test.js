import React from "react";
import { render, cleanup } from "@testing-library/react";
import CurrentDetailsComponent from "./CurrentDetailsComponent";
import { testStore } from "../../utils/test.util";

afterEach(cleanup);

test("should test current details component", () => {
  const store = testStore();
  const { asFragment } = render(<CurrentDetailsComponent store={store} />);
  expect(asFragment).toMatchSnapshot();
});
