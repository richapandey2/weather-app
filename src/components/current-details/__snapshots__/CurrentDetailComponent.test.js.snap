// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test current details component 1`] = `[Function]`;
